### PR TITLE
Implement PlayingData

### PR DESCRIPTION
**Pull request by clienthax**
Implementation of `REPLACE_WITH_DATA_MANIPULATOR`.
- [x] Registration of field getters and setters
- [x] Accurately used the appropriate `AbstractData` implementation

Implementation of `REPLACE_WITH_IMMUTABLE_MANIPULATOR`
- [x] Accurately extended the appropriate `AbstractImmutableData` implementation
- [x] Accurately instantiating final instance fields with an `ImmutableValue` counter part for any value getters
- [x] If necessary, creating a new `ImmutableValue` for a value that should not be cached, or, using the existing caching utils.

Implementation of the `DATA_MANIPULATOR_BUILDER`
- [x] In the `build()` method, checked if the container contained the keys required, if it doesn't, return `Optional.empty()`

Implementation of the `VALUE_PROCESSOR`(s):
- [x] Appropriately extended `AbstractSpongeValueProcessor`
- [x] Individual processors for each source type possible (one for `ItemStack`/`TileEntity`/`Entity` as necessary).

Registration:
- [x] Registered the `Key` correctly by `FIELD_NAME` in the `KeyRegistry`
- [x] Registered the `DataProcessor`s and `DataManipulatorBuilder` in `SpongeSerializationRegistry`
- [x] Registered the `ValueProcessor`s in the `SpongeSerializationRegistry`

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/builder/manipulator/mutable/entity/PlayingDataBuilder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public PlayingData create() {
+        return new SpongePlayingData();
+    }
+
+    @Override
+    public Optional<PlayingData> createFrom(DataHolder dataHolder) {
+        final boolean isPlaying;
+        if(supports(dataHolder)) {
```
Comment (Phroa) [inline src/main/java/org/spongepowered/common/data/builder/manipulator/mutable/entity/PlayingDataBuilder.java]: space after `if`

File: src/main/java/org/spongepowered/common/data/builder/manipulator/mutable/entity/PlayingDataBuilder.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            isPlaying = ((EntityVillager) dataHolder).isPlaying();
+            return Optional.of(new SpongePlayingData(isPlaying));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<PlayingData> build(DataView container) throws InvalidDataException {
+        if(container.contains(Keys.IS_PLAYING.getQuery())) {
```
Comment (Phroa) [inline src/main/java/org/spongepowered/common/data/builder/manipulator/mutable/entity/PlayingDataBuilder.java]: and here

File: src/main/java/org/spongepowered/common/data/key/KeyRegistry.java [hunk 3]
```diff
@@ -148,6 +148,8 @@ private static void generateKeyMap() {
         keyMap.put("placeable_blocks", makeSetKey(BlockType.class, of("CanPlaceOn")));
         keyMap.put("walking_speed", makeSingleKey(Double.class, Value.class, of("WalkingSpeed")));
         keyMap.put("flying_speed", makeSingleKey(Double.class, Value.class, of("FlyingSpeed")));
+        keyMap.put("is_playing", makeSingleKey(Boolean.class, Value.class, of("IsPlaying")));
+
```
Comment (Phroa) [inline src/main/java/org/spongepowered/common/data/key/KeyRegistry.java]: double space

File: src/main/java/org/spongepowered/common/data/builder/manipulator/mutable/entity/PlayingDataBuilder.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public PlayingData create() {
+        return new SpongePlayingData();
+    }
+
+    @Override
+    public Optional<PlayingData> createFrom(DataHolder dataHolder) {
+        final boolean isPlaying;
```
Comment (Phroa) [inline src/main/java/org/spongepowered/common/data/builder/manipulator/mutable/entity/PlayingDataBuilder.java]: this could be moved in scope

**Comment by hsyyid** [comment]
Nice job haxy

**Pull request closed without merging**